akka-actor-typed/src/main/scala/akka/actor/typed/internal/Restarter.scala

```diff
@@ -41,6 +34,7 @@ import akka.actor.typed.scaladsl.Behaviors
         case r: Restart ⇒
           new LimitedRestarter(initialBehavior, startedBehavior, r, retries = 0, deadline = OptionVal.None)
         case Resume(loggingEnabled) ⇒ new Resumer(startedBehavior, loggingEnabled)
+        case _: Stop                ⇒ initialBehavior
```

Comment:
I don't think this may not be good enough in case you wrap a supervision that catches a narrower exception with a supervision that captures a broader one?

```
supervise(
  supervise(targetBehavior(probe.ref))
        .onFailure[RuntimeException](stop)
).onFailure[Exception](restart)
```

Reply:
if `IllegalArgumentException` is thrown that shuld `stop`
if `IOException` is thrown it should `restart`

Reply:
Updated to have an explicit stopper, thanks @johanandren  :+1: